src/sql/workbench/contrib/notebook/test/browser/outputProcessor.test.ts

```diff
@@ -0,0 +1,136 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// import * as TypeMoq from 'typemoq';
+import * as assert from 'assert';
+import { nb } from 'azdata';
+import * as op from 'sql/workbench/contrib/notebook/browser/models/outputProcessor';
+import { JSONObject } from 'sql/workbench/services/notebook/common/jsonext';
+import { nbformat as nbformat } from 'sql/workbench/services/notebook/common/nbformat';
+import { getRandomString } from 'vs/editor/test/common/model/linesTextBuffer/textBufferAutoTestUtils';
+
+
+suite('OutputProcessor functions', function (): void {
+	suite('getData', async function (): Promise<void> {
+		for (const outputType of ['execute_result', 'display_data', 'update_display_data', 'stream', 'error'] as nbformat.OutputType[]) {
+			const text = getRandomString(1, 10);
+			const output: nb.ICellOutput = {
+				output_type: outputType
+			};
+			if (nbformat.isError(output)) {
+				for (const traceback of [undefined, [], [getRandomString(1, 10), getRandomString(1, 10)]]) {
+					for (const evalue of [undefined, getRandomString(1, 10)]) {
+						for (const ename of [text, '']) {
+							test(`test for outputType:${outputType}, ename:'${ename}', evalue:${evalue}, and traceback:${JSON.stringify(traceback)}`, async () => {
+								output.ename = ename;
+								output.evalue = evalue;
+								output.traceback = traceback;
+								const result = op.getData(output);
+								const tracedata = (traceback === undefined || traceback === []) ? undefined : traceback.join('\n');
+								let expectedData: JSONObject = {
+									'application/vnd.jupyter.stderr': undefined
+								};
+								if (tracedata) {
+									expectedData = {
+										'application/vnd.jupyter.stderr': tracedata
+									};
+								} else if (evalue) {
+									if (ename !== '') {
+										expectedData = {
+											'application/vnd.jupyter.stderr': `${output.ename}: ${output.evalue}`
+										};
+									} else {
+										expectedData = {
+											'application/vnd.jupyter.stderr': `${output.evalue}`
+										};
+									}
+
+								}
+								assert.deepEqual(result, <JSONObject>expectedData, `getData should return the expectedData:'${JSON.stringify(expectedData)}' object`);
```

Comment:
Well, I really hope if anyone has to debug failures in this test they have a deep knowledge of how this all works. Because there's absolutely no indication of why these are the expected values. Getting some comments in here explaining what's going on would be great.

Reply:
will do

Reply:
added comments